presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RuntimeReorderJoinSides.java

```diff
@@ -66,9 +76,35 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
             return Result.empty();
         }
 
-        StatsProvider statsProvider = context.getStatsProvider();
-        double leftOutputSizeInBytes = statsProvider.getStats(joinNode.getLeft()).getOutputSizeInBytes(joinNode.getLeft().getOutputVariables());
-        double rightOutputSizeInBytes = statsProvider.getStats(joinNode.getRight()).getOutputSizeInBytes(joinNode.getRight().getOutputVariables());
+        double leftOutputSizeInBytes = Double.NaN;
+        double rightOutputSizeInBytes = Double.NaN;
+        PlanNode actualLeft = context.getLookup().resolve(joinNode.getLeft());
+        PlanNode actualRight = context.getLookup().resolve(joinNode.getRight());
+        if (actualLeft instanceof TableScanNode && actualRight instanceof ExchangeNode && actualRight.getSources().size() == 1
```

Comment:
I would create a method `isUnmodifiedTableScan(PlanNode node)` (or some other name that makes sense) that traverses the tree with a visitor and returns true if the only nodes it encounters are exchange and TableScan node (and maybe project node that doesn't reduce the number of columns).  then change the condition to 

if(isUnmodifiedScan(left) && isUnmodifiedScan(right))
{

leftScan = getLeftScan(left)
rightScan = getRightScan (right)
}

that way we keep our conditions to what we need and don't rely on the plan looking a particular way (e.g. what if there was an extra local exchange or something).

Reply:
I guess how to refactor this part depends on our decision of the other comment.